File: manual/src/test/java/examples/java/LoggingReviewer.java
package examples.java;

import org.approvej.approve.PathProvider;
import org.approvej.review.ReviewResult;
import org.approvej.review.ReviewResultRecord;
import org.approvej.review.Reviewer;
import org.approvej.review.ReviewerProvider;
import org.jspecify.annotations.NonNull;

public class LoggingReviewer implements Reviewer, ReviewerProvider {

  @Override
  public ReviewResult apply(PathProvider pathProvider) {
    System.out.println("Received: " + pathProvider.receivedPath());
    System.out.println("Approved: " + pathProvider.approvedPath());
    return new ReviewResultRecord(false);
  }

  @Override
  public @NonNull String alias() {
    return "logging";
  }

  @Override
  public Reviewer create() {
    return new LoggingReviewer();
  }
}

File: manual/src/test/kotlin/examples/kotlin/LoggingReviewer.kt
package examples.kotlin

import org.approvej.approve.PathProvider
import org.approvej.review.ReviewResult
import org.approvej.review.ReviewResultRecord
import org.approvej.review.Reviewer
import org.approvej.review.ReviewerProvider

class LoggingReviewer : Reviewer, ReviewerProvider {
  override fun apply(pathProvider: PathProvider): ReviewResult {
    println("Received: " + pathProvider.receivedPath())
    println("Approved: " + pathProvider.approvedPath())
    return ReviewResultRecord(false)
  }

  override fun alias() = "logging"

  override fun create() = LoggingReviewer()
}
